lapce-ui/src/editor.rs

```diff
@@ -1787,6 +1789,127 @@ impl LapceEditor {
         }
     }
 
+    /// Checks if the cursor is on a bracket and highlights the matching bracket if there is one.
+    /// If the cursor is between brackets it highlights the enclosing brackets.
+    fn paint_bracket_highlight(
+        ctx: &mut PaintCtx,
+        data: &LapceEditorBufferData,
+        screen_lines: &ScreenLines,
+    ) {
+        if screen_lines.lines.is_empty() {
+            return;
+        }
+
+        let cursor_offset = data.editor.cursor.offset();
+
+        if let Some(char_at_cursor) = data.doc.buffer().char_at_offset(cursor_offset)
+        {
+            if is_bracket(char_at_cursor) {
+                if let Some(syntax) = data.doc.syntax() {
```

Comment:
I think this would look better if you retrieved the syntax without pushing code to the right. What I have in mind is something like this:

```rust
let syntax = if let Some(syntax) = data.doc.syntax() {
    syntax
} else {
    return;
};

if is_bracket(char_at_cursor) {...
```

This way we could remove a duplicated condition and use the left side of our monitors a bit more :)

Reply:
> This way we could remove a duplicated condition and use the left side of our monitors a bit more :)

Yeah, sorry. As a widescreen user I like to have code in the middle of my monitor to avoid a sore neck, but I get what you mean. I got carried away trying to avoid `.unwrap()`. :)

I'll fix this tonight 👍

Reply:
I think the `if let Some(_) = _ {}` syntax would look much nicer here

Reply:
either of those works

Reply:
Yeah, just some stylistic differences